src/graphics/Screen.cpp

```diff
@@ -2661,15 +2661,18 @@ int Screen::handleStatusUpdate(const meshtastic::Status *arg)
 }
 
 int Screen::handleTextMessage(const meshtastic_MeshPacket *packet)
+// If auto carousel is disabled, the focus doesn't chage on the incoming message
 {
-    if (showingNormalScreen) {
-        // Outgoing message
-        if (packet->from == 0)
-            setFrames(FOCUS_PRESERVE); // Return to same frame (quietly hiding the rx text message frame)
+    if (config.display.auto_screen_carousel_secs != 0) {
+        if (showingNormalScreen) {
+            // Outgoing message
+            if (packet->from == 0)
```

Comment:
Can we make this a ternary or something to clean up the nesting? 3 if statements down feels too nesty